### ISSUE TITLE
Help with public gateway docker

### ISSUE THREAD

**Issue by magaiverpr** [pglombardo/PasswordPusher#2498]
## Checklist

- [X] I've searched the project's [`issues`](https://github.com/pglombardo/PasswordPusher/issues?q=is%3Aissue).
- [X] I've searched the project's [`discussions`](https://github.com/pglombardo/PasswordPusher/discussions).

## ❓ Question

Hello,

I am trying to use the new container "Public Gateway", the new container mode.

Locally in my kubernetes, I am using password.mysite.local, and in Azure I am using password.mysite.com.

In my local kubernetes I am using the parameter PWP__OVERRIDE_BASE_URL: https://password.mysite.com, thats is sending correctly to the public gateway.

But when I access the public link, say that the link expired or not exist....

In azure purely used the container, without any ENV. There is more ENVs to configure both local and public dockers? I tried to find some documentations about this config, but didnt find any....

## 📎 Additional context

In my local kubernetes I am not using any database, only sending passwords, without any logs and login.

**Comment by pglombardo** [comment]
Hi @magaiverpr - The public gateway container and the pwpush container should have both the same settings and environment.  This way they both are connected to the same database and behave equally.

This needs to be clarified in the documentation admittedly.  I'll update the docs soon.

Could you try this and let me know?

**Comment by pglombardo** [comment]
I added a note to the documentation here:
https://docs.pwpush.com/docs/public-gateway/#deployment-architecture-example

**Comment by magaiverpr** [comment]
One question:

Is mandatory to use a DB to get the public gateway to work? I am using the ephemeral version, without any kind of DB (no need to store anything, even if the docker restart).

**Comment by pglombardo** [comment]
When using ephemeral, pwpush uses an SQLite3 database file at `/opt/PasswordPusher/db/db/sqlite3*`.  If you were to put this directory on a shared filesystem it would work fine for low traffic sites.

Unfortunately the two containers need to have a common knowledge source to properly function.

I hope this helps.

**Comment by pglombardo** [comment]
Hi @magaiverpr - have you made any progress on this issue?

You might be looking for a proxy instead.

**Comment by Viajaz** [comment]
I have the same issue, 404 on all links.

**Comment by Viajaz** [comment]
Having just posted that, I seem to have resolved my problem. Ensure the public and private pwpush containers are the same version/using the same tag and that your reverse proxy rules are configured correctly for the password URLs, the locale is now query parameter.

**Comment by Viajaz** [comment]
Ensure the public and private pwpush containers are the same version/using the same tag and that your reverse proxy rules are configured correctly for the password URLs, the locale is now query parameter and deletion URL has changed.

**Comment by pglombardo** [comment]
Good point - I've added this to the docs:
https://docs.pwpush.com/docs/public-gateway/#tips--best-practices

**Comment by magaiverpr** [comment]
Hello, sorry for the delay.

I had to stept back and rewrite my infraestructure. I wanted to put the gateway in a docker in Azure, but because the need for communication with a DB, I had to use my internal infraestructure to put de Pusher and put the Gateway in my DMZ.

Now its working as intended.

FYI: about the envs, I use only the customization/brand envs, not need for envs about consoants, separators, expire, etc.
